src/Util/Image.h

```diff
@@ -108,20 +119,63 @@ struct PointXy {
     {CARTA::PolarizationType::RL, 7}, {CARTA::PolarizationType::LR, 8}, {CARTA::PolarizationType::XX, 9}, {CARTA::PolarizationType::YY, 10},
-    {CARTA::PolarizationType::XY, 11}, {CARTA::PolarizationType::YX, 12}};
+    {CARTA::PolarizationType::XY, 11}, {CARTA::PolarizationType::YX, 12}, {CARTA::PolarizationType::Ptotal, 13},
+    {CARTA::PolarizationType::Plinear, 14}, {CARTA::PolarizationType::PFtotal, 15}, {CARTA::PolarizationType::PFlinear, 16},
+    {CARTA::PolarizationType::Pangle, 17}};
 
 static std::unordered_map<int, CARTA::PolarizationType> StokesTypes{{1, CARTA::PolarizationType::I}, {2, CARTA::PolarizationType::Q},
     {3, CARTA::PolarizationType::U}, {4, CARTA::PolarizationType::V}, {5, CARTA::PolarizationType::RR}, {6, CARTA::PolarizationType::LL},
     {7, CARTA::PolarizationType::RL}, {8, CARTA::PolarizationType::LR}, {9, CARTA::PolarizationType::XX}, {10, CARTA::PolarizationType::YY},
-    {11, CARTA::PolarizationType::XY}, {12, CARTA::PolarizationType::YX}};
+    {11, CARTA::PolarizationType::XY}, {12, CARTA::PolarizationType::YX}, {13, CARTA::PolarizationType::Ptotal},
+    {14, CARTA::PolarizationType::Plinear}, {15, CARTA::PolarizationType::PFtotal}, {16, CARTA::PolarizationType::PFlinear},
+    {17, CARTA::PolarizationType::Pangle}};
 
 static std::unordered_map<std::string, CARTA::PolarizationType> StokesStringTypes{{"I", CARTA::PolarizationType::I},
     {"Q", CARTA::PolarizationType::Q}, {"U", CARTA::PolarizationType::U}, {"V", CARTA::PolarizationType::V},
     {"RR", CARTA::PolarizationType::RR}, {"LL", CARTA::PolarizationType::LL}, {"RL", CARTA::PolarizationType::RL},
     {"LR", CARTA::PolarizationType::LR}, {"XX", CARTA::PolarizationType::XX}, {"YY", CARTA::PolarizationType::YY},
-    {"XY", CARTA::PolarizationType::XY}, {"YX", CARTA::PolarizationType::YX}};
+    {"XY", CARTA::PolarizationType::XY}, {"YX", CARTA::PolarizationType::YX}, {"Ptotal", CARTA::PolarizationType::Ptotal},
+    {"Plinear", CARTA::PolarizationType::Plinear}, {"PFtotal", CARTA::PolarizationType::PFtotal},
+    {"PFlinear", CARTA::PolarizationType::PFlinear}, {"Pangle", CARTA::PolarizationType::Pangle}};
 
 int GetStokesValue(const CARTA::PolarizationType& stokes_type);
 CARTA::PolarizationType GetStokesType(int stokes_value);
+bool ComputedStokes(int stokes);
+bool ComputedStokes(const std::string& stokes_type);
+
+// The struct StokesSrc is used to tell the file loader to get the original image interface, or get the computed stokes image interface.
+// The x, y, and z ranges from the StokesSrc indicate the range of image data to be calculated (for the new stokes type image).
+// We usually don't want to calculate the whole image data, because it spends a lot of time.
+// StokesSrc will bind casacore::Slicer or casacore::ImageRegion, because the coordinate of a computed stokes image is different from
+// the original image coordinate.
+struct StokesSrc {
```

Comment:
I think the full name improves readability: `StokesSource`. Shorter name are more variable-like:
```
src = StokesSource();
```